blog/2024-08-21-dotnet-sdk-v2.md

```diff
@@ -0,0 +1,192 @@
+      // code to run before flag evaluation
+    }
+// diff-add-block-end
+}
+```
+
+## Migration Steps
+
+### Application Authors
+
+#### Async suffixes and Cancellation Tokens
+
+Generally, application authors won't have to make many changes.
+
+As mentioned above, some async methods have been changed to confirm to .NET conventions.
+Evaluation methods now have the "Async" suffix, and accept an optional cancellationToken:
+
+```csharp
+// diff-remove
+    await client.GetBooleanValue("my-flag", false);
+// diff-add
+    await client.GetBooleanValueAsync("my-flag", false, cancellationToken);
+```
+
+The method for setting a provider has been updated similarly:
+
+```csharp
+// diff-remove
+    await Api.Instance.SetProvider(myProviderInstance);
+// diff-add
+    await Api.Instance.SetProviderAsync(myProviderInstance);
+```
+
+#### "Client Name" has Changed to "Domain"
+
+Parameters previously named "client name" are now named "domain".
+"[Domains](https://openfeature.dev/specification/glossary/#domain)" are a more powerful concept than "named clients", but in instances where named clients were previously used, function identically.
+No changes to consuming code are necessary these cases.
+
+### Provider Authors
+
+#### Provider Method Name Changes
+
+Provider authors must change their implementations in accordance with new methods names on the provider interface.
+Optionally, you can make use of the new cancellation token:
+
+```csharp
+// diff-remove-block-start
+
+    public override Task<ResolutionDetails<bool>> ResolveBooleanValue(
+                string flagKey,
+                bool defaultValue,
+                EvaluationContext? context = null)
+            {
+                // return boolean flag details
+            }
+// diff-remove-block-end
+// diff-add-block-start
+    public override Task<ResolutionDetails<bool>> ResolveBooleanValueAsync(
+                string flagKey,
+                bool defaultValue,
+                EvaluationContext? context = null,
+                CancellationToken cancellationToken = default)
+            {
+                // return boolean flag details
+            }
+// diff-add-block-end
+```
+
+#### No Need for Provider Status
+
+It's no longer necessary to define, update or expose provider status.
+If your provider requires initialization, simply define the optional `InitializeAsync` method:
+
+```csharp
+public class MyProvider : FeatureProvider
+{
+// diff-remove-block-start
+
+    private ProviderStatus _status = ProviderStatus.NotReady;        
+   
+    public override ProviderStatus GetStatus()
+    {
+        return _status;
+    }
+
+    public override Task Initialize(EvaluationContext context)
+// diff-remove-block-end
+// diff-add
+    public override Task InitializeAsync(EvaluationContext context)
+    {
+        // some async initialization
+    }
+}
+```
+
+For more details about this change, see our [previous blog post](https://openfeature.dev/blog/reconciling-with-state).
```

Comment:
```suggestion
For more details about this change, see our [previous blog post](/blog/reconciling-with-state).
```

Reply:
I think this will work. Please double-check before merging.

Reply:
But why is it better? This will break link when running it locally.